src/psyclone/psyir/backend/fortran.py

```diff
@@ -601,34 +601,54 @@ def gen_decls(self, symbol_table, args_allowed=True):
             which does not have a GlobalInterface or LocalInterface as this \
             is not supported by this backend.
         :raises VisitorError: if args_allowed is False and one or more \
-                              argument declarations exist in symbol_table.
-        :raises VisitorError: if any symbols representing variables (i.e. \
-            not kind parameters) without an explicit declaration or 'use' \
-            are encountered.
+            argument declarations exist in symbol_table.
+        :raises VisitorError: if any symbols representing variables without \
+            a (local) explicit declaration or 'use' are encountered and there \
```

Comment:
I think the "(local)" should be removed as get_unresolved_datasymbols checks at any level.

Also, can the sentence be simplyfied? Maybe "if any symbols without an explicit declaration is found and there are no wildcard imports."

Reply:
I was getting a bit confused. I've sorted myself out and reworded :-)